usr/share/rear/layout/save/GNU/Linux/260_crypt_layout.sh

```diff
@@ -40,42 +37,45 @@ while read target_name junk ; do
         Log "Skipping $target_name (could not get its device in /sys/block/$sysfs_device/slaves/)"
         continue
     fi
-    
-    if ! cryptsetup isLuks $source_device ; then
+
+    if ! blkid -p -o export $source_device >$TMP_DIR/blkid.output ; then
+        LogPrintError "Error: Cannot get attributes for $target_name ('blkid -p -o export  $source_device' failed)"
+        continue
+    fi
+
+    if ! grep -q "TYPE=crypto_LUKS" $TMP_DIR/blkid.output ; then
         Log "Skipping $target_name (its $source_device is not a LUKS device)"
         continue
     fi
 
+    # Detect LUKS version
+    version=$( grep "VERSION" $TMP_DIR/blkid.output | cut -d= -f2 )
+    if !( test $version = "1" || test $version = "2") ; then
```

Comment:
The condition `!( ...)` will not work as intended
because "man bash" reads
```
If the extglob shell option is enabled
...
 !(pattern-list)
 Matches anything except one of the given patterns
```
and usr/sbin/rear has the extglob shell option enabled
https://github.com/rear/rear/blob/master/usr/sbin/rear#L314

Reply:
Sigh - I needed three commits to now hopefully have that fixed.
Sometimes bash scripting is so weird.